### PR TITLE
SAGA: Support ITE windows demo 1 and 2

### PR DESCRIPTION
**Pull request by phcoder**

### CODE REVIEW

**Comment by sev-** [comment]
This doesn't link:

```C
/usr/bin/ld: engines/drascula/libdrascula.a(resource.o): in function `Drascula::ArchiveMan::registerArchive(Common::String const&, int)':
/home/runner/work/scummvm/scummvm/engines/drascula/resource.cpp:32: undefined reference to `Common::makeArjArchive(Common::String const&)'
```

**Comment by phcoder** [comment]
> This doesn't link:
> 
> ```c
> /usr/bin/ld: engines/drascula/libdrascula.a(resource.o): in function `Drascula::ArchiveMan::registerArchive(Common::String const&, int)':
> /home/runner/work/scummvm/scummvm/engines/drascula/resource.cpp:32: undefined reference to `Common::makeArjArchive(Common::String const&)'
> ```

fixed

File: common/compression/unarj.h [hunk 1]
```diff
@@ -49,12 +49,12 @@ class Archive;
  *
  * May return 0 in case of a failure.
  */
-Archive *makeArjArchive(const String &name);
+Archive *makeArjArchive(const String &name, bool flattenTree=false);
```
Comment (bluegr) [inline common/compression/unarj.h:52]:
```suggestion
Archive *makeArjArchive(const String &name, bool flattenTree = false);
```

File: common/compression/unarj.h [hunk 2]
```diff
@@ -49,12 +49,12 @@ class Archive;
...
 
 /**
  * Similar to previous but for multi-volume archives
  */
-Archive *makeArjArchive(const Array<String> &names);
+Archive *makeArjArchive(const Array<String> &names, bool flattenTree=false);
```
Comment (bluegr) [inline common/compression/unarj.h:57]:
```suggestion
Archive *makeArjArchive(const Array<String> &names, bool flattenTree = false);
```

**Comment by phcoder** [comment]
@bluegr merged your white space fixes into the stack

**Comment by sev-** [comment]
Thanks!

**Pull request merged**